src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -82,8 +93,53 @@ private ConcatImmutableArray<CodeRefactoringProvider> GetProviders(Document docu
 
         public async Task<bool> HasRefactoringsAsync(
             Document document,
-            TextSpan state,
+            TextSpan textSpan,
+            TextSpan? pastedTextSpan,
             CancellationToken cancellationToken)
+        {
+            var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
+            if (client is object)
+            {
+                var result = await client.TryRunRemoteAsync<bool>(
+                    WellKnownServiceHubServices.CodeAnalysisService,
+                    nameof(IRemoteCodeRefactoringService.HasRefactoringsAsync),
+                    document.Project.Solution,
+                    new object?[]
+                    {
+                        document.Id,
+                        textSpan,
+                        pastedTextSpan,
+                    },
+                    callbackTarget: new CodeRefactoringServiceCallback(this),
```

Comment:
tbh.... i don't get what this callback is for.  i don't eally see anyone consuming it on the remote side.  could you clarify?